packages/plugin-devtools/test/devtools.test.ts

```diff
@@ -0,0 +1,735 @@
+            await longNextTick();
+            window.dispatchEvent(new MouseEvent('mousemove', end));
+            await longNextTick();
+            navbar.dispatchEvent(new MouseEvent('mouseup', end));
+            await nextTick();
+            navbar.dispatchEvent(new MouseEvent('click', end));
+            await nextTick();
+            expect(devtools.classList.contains('closed')).to.equal(false);
+        });
+        it('should close the devTools', async () => {
+            const navbar = devtools.querySelector('devtools-navbar');
+            navbar.dispatchEvent(new MouseEvent('mousedown', { bubbles: true }));
+            await nextTick();
+            navbar.dispatchEvent(new MouseEvent('mouseup', { bubbles: true }));
+            await nextTick();
+            navbar.dispatchEvent(new MouseEvent('click', { bubbles: true }));
+            await nextTick();
+            expect(devtools.classList.contains('closed')).to.equal(true);
+        });
+        it('should open the devTools when click on button "Commands"', async () => {
+            const button = devtools.querySelector('devtools-navbar button:nth-child(2)');
+            button.dispatchEvent(new MouseEvent('mousedown', { bubbles: true }));
+            await nextTick();
+            button.dispatchEvent(new MouseEvent('mouseup', { bubbles: true }));
+            await nextTick();
+            button.dispatchEvent(new MouseEvent('click', { bubbles: true }));
+            await longNextTick();
+            expect(devtools.classList.contains('closed')).to.equal(false);
+        });
+    });
+    describe('InspectorComponent', () => {
+        it('should change panel to "Inspector" with top button', async () => {
+            const button = devtools.querySelector('devtools-navbar button');
+            expect(button.textContent).to.equal('Inspector');
+            expect(button.classList.contains('selected')).to.equal(false, 'button is not selected');
+
+            button.dispatchEvent(new MouseEvent('click', { bubbles: true }));
+            await longNextTick();
+
+            expect(button.classList.contains('selected')).to.equal(true, 'button is selected');
+            expect(!!devtools.querySelector('devtools-panel.active devtools-tree')).to.equal(true);
+        });
+        it('should display the VNode tab by default in info', async () => {
+            const button = devtools.querySelector('devtools-info devtools-navbar button');
+            expect(button.textContent).to.equal('VNode');
+            expect(button.classList.contains('selected')).to.equal(true, 'button is selected');
+        });
+        it('should display the root by default in info', async () => {
+            const about = devtools.querySelector('devtools-info .about');
+            const aResult =
+                '<div class="about">' +
+                '<span class="type">VNode</span> ROOT<span class="id">0</span>' +
+                '</div>';
```

Comment:
Could we test this without being 100% coupled with the DOM structure of the devtools ? Because that is subject to change.

Reply:
Yes, and if that changes it is normal to update the tests. this has already allowed an error to be detected (used in Antoine's branch)